pallets/creditcoin/src/helpers.rs

```diff
@@ -189,7 +189,7 @@ fn extract_public_key_eth_sign<T: Config>(
 				sp_core::ecdsa::Public::from_raw(public_key),
 			) {
 				Some(s) => Ok(s),
-				None => Err(Error::EthSignExternalAddressGenerationFailed),
+				None => Err(Error::MalformedExternalAddress),
```

Comment:
IMO this is wrong.  You are changing the test target in order to accommodate expected results inside the tests instead of adapting the tests to match reality. 

`Error::MalformedExternalAddress` is currently used in a different context, for example:

```
                                        ensure!(
                                                helpers::address_is_well_formed(&blockchain, &address),
                                                Error::<T>::MalformedExternalAddress
                                        );
```

Reply:
I was attempting to use a single line of code to cover two error scenarios which is why I changed it the way I did. But you are right that is definitely not the best style so I broke out the two errors I was worried about into explicit calls at the start of the extrinsic.


```rust
ensure!(
    helpers::blockchain_is_supported(&blockchain),
    Error::<T>::UnsupportedBlockchain
);

ensure!(
    helpers::address_is_well_formed(&blockchain, &address),
    Error::<T>::MalformedExternalAddress
);
```

I then added an explicit `register_address_v2_should_error_with_unsupported_blockchain` unit test and returned the errors to the more specific ones that they were before. Technically right now those errors are unreachable as the way they can fail in generate_external_address is through being unsupported or malformed but its definitely more descriptive and adds room to grow.